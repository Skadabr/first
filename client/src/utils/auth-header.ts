import axios from "axios";

export default function setAuthHeader(token?: string) {
  if (token) {
    axios.defaults.headers.common.authorization = `Bearer ${token}`;
  } else {
    delete axios.defaults.headers.common.authorization;
  }
}
