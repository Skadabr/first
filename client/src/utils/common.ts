export function createNumberSequence(n) {
  const res = [];
  for (let i = 0; i < n; i++) res.push(i);
  return res;
}

