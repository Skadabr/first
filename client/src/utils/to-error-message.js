export default function toErrorMessage(e) {
  if (e.response && e.response.data) {
    if (e.response.data.error) {
      throw new Error(e.response.data.error.message);
    } else {
      throw new Error("Bad request");
    }
  }
  throw e;
}
