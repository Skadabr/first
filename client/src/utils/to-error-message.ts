export default function toErrorMessage(e: any) {
  if (e.response && e.response.data) {
    if (e.response.data.error) {
      throw new Error(e.response.data.error.message);
    } else {
      throw new Error("Bad request");
    }
  }
  throw e;
}
