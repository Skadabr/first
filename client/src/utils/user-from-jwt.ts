export default function() {
  const token = localStorage.user_jwt;
}
