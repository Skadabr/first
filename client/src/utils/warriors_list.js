import { PAWN, OFFICER } from "../constants";

module.exports = [
  {
    type: PAWN,
    health: 6,
    damage: 1,
    price: 1
  },
  {
    type: OFFICER,
    health: 4,
    damage: 2,
    price: 2
  }
];
