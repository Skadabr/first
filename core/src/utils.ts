export class UnitSet {
  private units: any[];

  constructor() {
    this.units = [];
  }

  add(unit) {
    this.units = [...this.units.filter(unit._id), unit];
  }

  toArray() {
    return this.units;
  }
}

export function createNumberSequence(n: number): number[] {
  const res: number[] = [];
  for (let i = 0; i < n; i++) res.push(i);
  return res;
}

export function generateID() {
  return Math.random().toString();
}

export function createUniqueArray(arr) {
  return [...new Set(arr)];
}
