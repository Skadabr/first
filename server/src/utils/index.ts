const { NODE_ENV } = process.env;

export function getErrorMessage(err) {
  if (NODE_ENV === "development") {
    return err.stack;
  } else {
    return err.message;
  }
}
