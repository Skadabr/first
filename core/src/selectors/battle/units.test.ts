import dcopy from "deep-copy";

import { baseState } from "../../__tests__/fixtures/state-fixture";
import {
  getEnemyHero,
  getEnemyMinionsByUserId,
  getEnemyUnitIdsByUserId,
  getEnemyUnits,
  getEnemyUnitsByUserId,
  getHero,
  getMinionsByUserId,
  getPlayerUnits,
  getUnitById,
  getUnitFriendlyMinions,
  getUnitFriendlyUnits,
  getUnitIdsByUserId,
  getUnitsByTargetingScope,
  getUnitsByUserId
} from "./units";
import { EffectTargetingScope, UnitTypes } from "../../index";
import createUnit from "../../unit/fabric";
import {addTestUnit} from "../../__tests__/utils";

describe("Units selectors", () => {
  let state;

  beforeAll(function initState() {
    state = dcopy(baseState);

    addTestUnit(state, "pl_pawn1",  UnitTypes.Pawn,"player");
    addTestUnit(state, "pl_pawn2",  UnitTypes.Pawn,"player");
    addTestUnit(state, "pl_pawn3",  UnitTypes.Pawn,"player");
    addTestUnit(state, "e_pawn1",  UnitTypes.Pawn,"enemy");
  });

  describe("getUnitById", () => {
    it("should return a unit", function() {
      const unit = getUnitById(state, "pl_pawn1");
      expect(unit).toMatchSnapshot();
    });
  });

  describe("getUnitsByUserId", () => {
    it("should return list of `player` users", function() {
      const units = getUnitsByUserId(state, "player");
      expect(units).toMatchSnapshot();
    });
  });

  describe("getMinionsByUserId", () => {
    it("should return list of `player` users", function() {
      const units = getMinionsByUserId(state, "player");
      expect(units).toMatchSnapshot();
    });
  });

  describe("getHero", () => {
    describe("when there are many hero for one player", () => {
      let newState;

      beforeEach(function() {
        newState = dcopy(state);
        newState.battle.units.push({
          ...createUnit(UnitTypes.Pawn, "player"),
          _id: "pl_hero2",
          hero: true
        });
      });

      it("should throw an error", function() {
        expect(() => {
          getHero(newState, "player");
        }).toThrow("There should be only one hero");
      });
    });

    describe("when only one hero", () => {
      it("should return unit with `hero` attribute true", function() {
        const hero = getHero(state, "player");
        expect(hero.hero).toBe(true);
      });
    });
  });

  describe("getEnemyUnitsByUserId", () => {
    it("should return all enemy units of player which has passed id", function() {
      const units = getEnemyUnitsByUserId(state, "player");
      expect(units).toMatchSnapshot();
    });
  });

  describe("getEnemyMinionsByUserId", () => {
    it("should return all enemy minions of player which has passed id", function() {
      const units = getEnemyMinionsByUserId(state, "player");
      expect(units).toMatchSnapshot();
    });
  });

  describe("getEnemyHero", () => {
    describe("when there are many hero for one player", () => {
      let newState;

      beforeEach(function() {
        newState = dcopy(state);
        newState.battle.units.push({
          ...createUnit(UnitTypes.Pawn, "enemy"),
          _id: "e_hero2",
          hero: true
        });
      });

      it("should throw an error", function() {
        expect(() => {
          getEnemyHero(newState, "player");
        }).toThrow("There should be only one hero");
      });
    });

    describe("when only one hero", () => {
      it("should return unit with `hero` attribute true", function() {
        const hero = getEnemyHero(state, "player");
        expect(hero.hero).toBe(true);
      });
    });
  });

  describe("getUnitIdsByUserId", () => {
    it("should return list of all ids of `player`", function() {
      const ids = getUnitIdsByUserId(state, "player");
      expect(ids).toMatchSnapshot();
    });
  });

  describe("getEnemyUnitIdsByUserId", () => {
    it("should return list of all ids of `enemy`", function() {
      const ids = getEnemyUnitIdsByUserId(state, "player");
      expect(ids).toMatchSnapshot();
    });
  });

  describe("getPlayerUnits", () => {
    it("should return units of current player", function() {
      const units = getPlayerUnits(state);
      expect(units).toMatchSnapshot();
    });
  });

  describe("getEnemyUnits", () => {
    it("should return units of enemy of the current user", function() {
      const units = getEnemyUnits(state);
      expect(units).toMatchSnapshot();
    });
  });

  describe("getUnitFriendlyMinions", () => {
    it("should return list of friendly minions", function() {
      const units = getUnitFriendlyMinions(state, "pl_pawn1");
      expect(units).toMatchSnapshot();
    });
  });

  describe("getUnitFriendlyUnits", () => {
    it("should return list of all friendly units", function() {
      const units = getUnitFriendlyUnits(state, "pl_pawn1");
      expect(units).toMatchSnapshot();
    });
  });

  describe("getUnitsByTargetingScope", () => {
    const sourceId = "pl_pawn1";

    test("get other friendly minions", function() {
      const units = getUnitsByTargetingScope(
        state,
        sourceId,
        EffectTargetingScope.OtherFriendlyMinions
      );
      expect(units).toMatchSnapshot();
    });

    test("get all friendly minions", function() {
      const units = getUnitsByTargetingScope(
        state,
        sourceId,
        EffectTargetingScope.AllFriendlyMinions
      );
      expect(units).toMatchSnapshot();
    });

    test("get all friendly units", function() {
      const units = getUnitsByTargetingScope(
        state,
        sourceId,
        EffectTargetingScope.AllFriendlyUnits
      );
      expect(units).toMatchSnapshot();
    });

    test("get all enemy units", function() {
      const units = getUnitsByTargetingScope(
        state,
        sourceId,
        EffectTargetingScope.AllEnemyUnits
      );
      expect(units).toMatchSnapshot();
    });

    test("get all enemy minions", function() {
      const units = getUnitsByTargetingScope(
        state,
        sourceId,
        EffectTargetingScope.AllEnemyMinions
      );
      expect(units).toMatchSnapshot();
    });

    test("get all units", function() {
      const units = getUnitsByTargetingScope(
        state,
        sourceId,
        EffectTargetingScope.AllUnits
      );
      expect(units).toMatchSnapshot();
    });

    test("get local unit", function() {
      const units = getUnitsByTargetingScope(
        state,
        sourceId,
        EffectTargetingScope.Local
      );
      expect(units).toMatchSnapshot();
    });
  });
});
