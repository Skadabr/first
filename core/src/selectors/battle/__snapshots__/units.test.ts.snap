// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getEnemyMinionsByUserId should return all enemy minions of player which has passed id 1`] = `
Array [
  Object {
    "_id": "e_pawn3",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [],
    "health": 6,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "enemy",
    "type": "PAWN",
  },
]
`;

exports[`getEnemyUnitIdsByUserId should return list of all ids of \`enemy\` 1`] = `
Array [
  "e_hero",
  "e_pawn3",
]
`;

exports[`getEnemyUnits should return units of enemy of the current user 1`] = `
Array [
  Object {
    "_id": "e_hero",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [],
    "health": 6,
    "hero": true,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "enemy",
    "type": "PAWN",
  },
  Object {
    "_id": "e_pawn3",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [],
    "health": 6,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "enemy",
    "type": "PAWN",
  },
]
`;

exports[`getEnemyUnitsByUserId should return all enemy units of player which has passed id 1`] = `
Array [
  Object {
    "_id": "e_hero",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [],
    "health": 6,
    "hero": true,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "enemy",
    "type": "PAWN",
  },
  Object {
    "_id": "e_pawn3",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [],
    "health": 6,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "enemy",
    "type": "PAWN",
  },
]
`;

exports[`getMinionsByUserId should return list of \`player\` users 1`] = `
Array [
  Object {
    "_id": "pl_pawn1",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [],
    "health": 6,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "player",
    "type": "PAWN",
  },
  Object {
    "_id": "pl_pawn2",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [
      Object {
        "impact": "HEALTH",
        "ownerId": "pl_pawn2",
        "targetingScope": "OTHER_FRIENDLY_MINIONS",
        "type": "HEALTH",
        "value": 2,
      },
    ],
    "health": 6,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "player",
    "type": "PAWN",
  },
  Object {
    "_id": "pl_pawn3",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [
      Object {
        "impact": "ATTACK",
        "ownerId": "pl_pawn3",
        "targetingScope": "OTHER_FRIENDLY_MINIONS",
        "type": "ATTACK",
        "value": 1,
      },
    ],
    "health": 6,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "player",
    "type": "PAWN",
  },
]
`;

exports[`getPlayerUnits should return units of current player 1`] = `
Array [
  Object {
    "_id": "pl_hero",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [],
    "health": 6,
    "hero": true,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "player",
    "type": "PAWN",
  },
  Object {
    "_id": "pl_pawn1",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [],
    "health": 6,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "player",
    "type": "PAWN",
  },
  Object {
    "_id": "pl_pawn2",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [
      Object {
        "impact": "HEALTH",
        "ownerId": "pl_pawn2",
        "targetingScope": "OTHER_FRIENDLY_MINIONS",
        "type": "HEALTH",
        "value": 2,
      },
    ],
    "health": 6,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "player",
    "type": "PAWN",
  },
  Object {
    "_id": "pl_pawn3",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [
      Object {
        "impact": "ATTACK",
        "ownerId": "pl_pawn3",
        "targetingScope": "OTHER_FRIENDLY_MINIONS",
        "type": "ATTACK",
        "value": 1,
      },
    ],
    "health": 6,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "player",
    "type": "PAWN",
  },
]
`;

exports[`getUnitById should return a unit 1`] = `
Object {
  "_id": "pl_pawn1",
  "attack": 1,
  "cost": 1,
  "counterEffects": Array [],
  "effects": Array [],
  "health": 6,
  "moves": 0,
  "name": "Pawn",
  "ownerId": "player",
  "type": "PAWN",
}
`;

exports[`getUnitFriendlyMinions should return list of friendly minions 1`] = `
Array [
  Object {
    "_id": "pl_pawn2",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [
      Object {
        "impact": "HEALTH",
        "ownerId": "pl_pawn2",
        "targetingScope": "OTHER_FRIENDLY_MINIONS",
        "type": "HEALTH",
        "value": 2,
      },
    ],
    "health": 6,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "player",
    "type": "PAWN",
  },
  Object {
    "_id": "pl_pawn3",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [
      Object {
        "impact": "ATTACK",
        "ownerId": "pl_pawn3",
        "targetingScope": "OTHER_FRIENDLY_MINIONS",
        "type": "ATTACK",
        "value": 1,
      },
    ],
    "health": 6,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "player",
    "type": "PAWN",
  },
]
`;

exports[`getUnitFriendlyUnits should return list of all friendly units 1`] = `
Array [
  Object {
    "_id": "pl_hero",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [],
    "health": 6,
    "hero": true,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "player",
    "type": "PAWN",
  },
  Object {
    "_id": "pl_pawn2",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [
      Object {
        "impact": "HEALTH",
        "ownerId": "pl_pawn2",
        "targetingScope": "OTHER_FRIENDLY_MINIONS",
        "type": "HEALTH",
        "value": 2,
      },
    ],
    "health": 6,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "player",
    "type": "PAWN",
  },
  Object {
    "_id": "pl_pawn3",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [
      Object {
        "impact": "ATTACK",
        "ownerId": "pl_pawn3",
        "targetingScope": "OTHER_FRIENDLY_MINIONS",
        "type": "ATTACK",
        "value": 1,
      },
    ],
    "health": 6,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "player",
    "type": "PAWN",
  },
]
`;

exports[`getUnitIdsByUserId should return list of all ids of \`player\` 1`] = `
Array [
  "pl_hero",
  "pl_pawn1",
  "pl_pawn2",
  "pl_pawn3",
]
`;

exports[`getUnitsByTargetingScope get all enemy minions 1`] = `
Array [
  Object {
    "_id": "e_pawn3",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [],
    "health": 6,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "enemy",
    "type": "PAWN",
  },
]
`;

exports[`getUnitsByTargetingScope get all enemy units 1`] = `
Array [
  Object {
    "_id": "e_hero",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [],
    "health": 6,
    "hero": true,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "enemy",
    "type": "PAWN",
  },
  Object {
    "_id": "e_pawn3",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [],
    "health": 6,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "enemy",
    "type": "PAWN",
  },
]
`;

exports[`getUnitsByTargetingScope get all friendly minions 1`] = `
Array [
  Object {
    "_id": "pl_pawn1",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [],
    "health": 6,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "player",
    "type": "PAWN",
  },
  Object {
    "_id": "pl_pawn2",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [
      Object {
        "impact": "HEALTH",
        "ownerId": "pl_pawn2",
        "targetingScope": "OTHER_FRIENDLY_MINIONS",
        "type": "HEALTH",
        "value": 2,
      },
    ],
    "health": 6,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "player",
    "type": "PAWN",
  },
  Object {
    "_id": "pl_pawn3",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [
      Object {
        "impact": "ATTACK",
        "ownerId": "pl_pawn3",
        "targetingScope": "OTHER_FRIENDLY_MINIONS",
        "type": "ATTACK",
        "value": 1,
      },
    ],
    "health": 6,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "player",
    "type": "PAWN",
  },
]
`;

exports[`getUnitsByTargetingScope get all friendly units 1`] = `
Array [
  Object {
    "_id": "pl_hero",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [],
    "health": 6,
    "hero": true,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "player",
    "type": "PAWN",
  },
  Object {
    "_id": "pl_pawn1",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [],
    "health": 6,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "player",
    "type": "PAWN",
  },
  Object {
    "_id": "pl_pawn2",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [
      Object {
        "impact": "HEALTH",
        "ownerId": "pl_pawn2",
        "targetingScope": "OTHER_FRIENDLY_MINIONS",
        "type": "HEALTH",
        "value": 2,
      },
    ],
    "health": 6,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "player",
    "type": "PAWN",
  },
  Object {
    "_id": "pl_pawn3",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [
      Object {
        "impact": "ATTACK",
        "ownerId": "pl_pawn3",
        "targetingScope": "OTHER_FRIENDLY_MINIONS",
        "type": "ATTACK",
        "value": 1,
      },
    ],
    "health": 6,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "player",
    "type": "PAWN",
  },
]
`;

exports[`getUnitsByTargetingScope get all units 1`] = `
Array [
  Object {
    "_id": "pl_hero",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [],
    "health": 6,
    "hero": true,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "player",
    "type": "PAWN",
  },
  Object {
    "_id": "pl_pawn1",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [],
    "health": 6,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "player",
    "type": "PAWN",
  },
  Object {
    "_id": "pl_pawn2",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [
      Object {
        "impact": "HEALTH",
        "ownerId": "pl_pawn2",
        "targetingScope": "OTHER_FRIENDLY_MINIONS",
        "type": "HEALTH",
        "value": 2,
      },
    ],
    "health": 6,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "player",
    "type": "PAWN",
  },
  Object {
    "_id": "pl_pawn3",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [
      Object {
        "impact": "ATTACK",
        "ownerId": "pl_pawn3",
        "targetingScope": "OTHER_FRIENDLY_MINIONS",
        "type": "ATTACK",
        "value": 1,
      },
    ],
    "health": 6,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "player",
    "type": "PAWN",
  },
  Object {
    "_id": "e_hero",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [],
    "health": 6,
    "hero": true,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "enemy",
    "type": "PAWN",
  },
  Object {
    "_id": "e_pawn3",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [],
    "health": 6,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "enemy",
    "type": "PAWN",
  },
]
`;

exports[`getUnitsByTargetingScope get local unit 1`] = `
Array [
  Object {
    "_id": "pl_pawn1",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [],
    "health": 6,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "player",
    "type": "PAWN",
  },
]
`;

exports[`getUnitsByTargetingScope get other friendly minions 1`] = `
Array [
  Object {
    "_id": "pl_pawn2",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [
      Object {
        "impact": "HEALTH",
        "ownerId": "pl_pawn2",
        "targetingScope": "OTHER_FRIENDLY_MINIONS",
        "type": "HEALTH",
        "value": 2,
      },
    ],
    "health": 6,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "player",
    "type": "PAWN",
  },
  Object {
    "_id": "pl_pawn3",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [
      Object {
        "impact": "ATTACK",
        "ownerId": "pl_pawn3",
        "targetingScope": "OTHER_FRIENDLY_MINIONS",
        "type": "ATTACK",
        "value": 1,
      },
    ],
    "health": 6,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "player",
    "type": "PAWN",
  },
]
`;

exports[`getUnitsByUserId should return list of \`player\` users 1`] = `
Array [
  Object {
    "_id": "pl_hero",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [],
    "health": 6,
    "hero": true,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "player",
    "type": "PAWN",
  },
  Object {
    "_id": "pl_pawn1",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [],
    "health": 6,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "player",
    "type": "PAWN",
  },
  Object {
    "_id": "pl_pawn2",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [
      Object {
        "impact": "HEALTH",
        "ownerId": "pl_pawn2",
        "targetingScope": "OTHER_FRIENDLY_MINIONS",
        "type": "HEALTH",
        "value": 2,
      },
    ],
    "health": 6,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "player",
    "type": "PAWN",
  },
  Object {
    "_id": "pl_pawn3",
    "attack": 1,
    "cost": 1,
    "counterEffects": Array [],
    "effects": Array [
      Object {
        "impact": "ATTACK",
        "ownerId": "pl_pawn3",
        "targetingScope": "OTHER_FRIENDLY_MINIONS",
        "type": "ATTACK",
        "value": 1,
      },
    ],
    "health": 6,
    "moves": 0,
    "name": "Pawn",
    "ownerId": "player",
    "type": "PAWN",
  },
]
`;
